### PR TITLE
Update colors to look okay for both white and black terms. Fix wording

### PR DESCRIPTION
**Pull request by sophiajt**

### CODE REVIEW

File: src/libsyntax/errors/emitter.rs [hunk 1]
```diff
@@ -361,16 +361,28 @@ fn print_diagnostic(dst: &mut Destination,
...
-        dst.reset_attrs()?;
-    }
 
-    dst.start_attr(term::Attr::Bold)?;
-    write!(dst, "{}: ", lvl.to_string())?;
-    write!(dst, "{}", msg)?;
-    dst.reset_attrs()?;
+
+    match lvl {
+        Level::Error => {
```
Comment (nikomatsakis) [inline src/libsyntax/errors/emitter.rs]: hmm, one concern I have is that if the "top-level" error is a warning, this won't get bolded. I put in some logic to emit newlines only before the _start_ of an error -- perhaps we should modify that to push down a boolean flag, and use that flag as the criteria here?

Comment (sophiajt) [inline src/libsyntax/errors/emitter.rs]: Ah, good point, I missed that.  Right, the important piece is to handle the top line differently.  Do we want a separate header renderer?

Comment (sophiajt) [inline src/libsyntax/errors/emitter.rs]: I can also just special case warning and error for now...

Comment (nikomatsakis) [inline src/libsyntax/errors/emitter.rs]: That's true, special casing warning is probably ok. There are some cases where we have embedded warnings though. If we can do the header thing I think it'd be _better_ (doesn't seem ... that hard).

Comment (nikomatsakis) [inline src/libsyntax/errors/emitter.rs]: But I'd be fine with leaving it as a future to do item.

Comment (sophiajt) [inline src/libsyntax/errors/emitter.rs]: Special case for now, then have header_render as a todo sounds good

**Pull request merged**